core/src/models/Group.ts

```diff
@@ -37,7 +37,7 @@ export const GROUP_RULE_LIMIT = 10;
 const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
 
 export interface GroupRuleWithKey {
-  key: string;
+  key?: string;
```

Comment:
This doesn't feel like a good decision, but I also didn't want to make a new type. However, when we're building an object of rules to be written to file, we don't need the key to come along, and I didn't want to bring anything that the developer isn't going to care about.